### ISSUE TITLE
The initialize method makes a copy of config file in input_dir 

### ISSUE THREAD

**Issue by SarahAlidoost** [eWaterCycle/grpc4bmi#68]
By default, the [initialize method](https://grpc4bmi.readthedocs.io/en/latest/grpc4bmi.html#grpc4bmi.bmi_client_docker.BmiClientDocker.initialize) makes a copy of [config file in input_dir](https://github.com/eWaterCycle/grpc4bmi/blob/13ca84fc66d0d866f24f4dd606bb8ae0fcd911e8/grpc4bmi/utils.py#L28). This is a problem where there is no writing permission in input_dir, for example, when running a model on ewatercycle Jupyter machine. The input_dir mainly contains several fixed files used as input for a model whereas configuration file contains temporary settings. The changed config file does not need to be copied in the input_dir.

**Comment by goord** [comment]
Ok, but that will mean that **all** paths in the config file will have to absolute

**Comment by goord** [comment]
...which means the config file is not portable anymore...

**Comment by goord** [comment]
So the idea is that the ```input_dir``` is actually writable by the user: it contains data that one wants to experiment with, e.g. the configuration file and possible sub-configurations files. BTW the input config file is only copied into this directory if it isn't there already (it needs to be visible for the model inside the container).

Data on read-only data folders are supported by adding extra volumes to the client constructor.

Note that all this means that the config files have to have their paths correctly set up: paths to input files should be either ``/data/input/...`` or an extra mounted volume, output should be ``/data/output``

**Comment by SarahAlidoost** [comment]
@goord thank you for the explanations. So, I think that adding extra volumes (issues #65 and #66) will solve this problem.

**Comment by RolfHut** [comment]
We (@jeromaerts and me) ran into this problem as users when constructing notebooks for the technical paper.  I agree with @goord that it is not desirable to use absolute paths in the config file, because this makes porting to different systems a problem. At the same time, for many applications you don't have any rights on the directory that contains the big input files your model needs. 

Is it a solution to have a third mount next to input_dir and output_dir such as 'run_dir'? Or maybe write the ini file to the output_dir, so the user has a copy of the ini file that was used to generate that output by default?

**Comment by SarahAlidoost** [comment]
@RolfHut and @jeromaerts For mounting, the grpc4bmi client modules support the option `extra_volumes`, see [grpc4bmi.bmi_client_docker](https://grpc4bmi.readthedocs.io/en/latest/grpc4bmi.html#module-grpc4bmi.bmi_client_docker)  and [grpc4bmi.bmi_client_singularity](https://grpc4bmi.readthedocs.io/en/latest/grpc4bmi.html#module-grpc4bmi.bmi_client_singularity). However, using this option depends on the model structure. For example, we could set this option for **lisflood**, please see `cell #8` in [02_lisflood_testrun.ipynb](https://github.com/eWaterCycle/era5-comparison/blob/master/lisflood/02_lisflood_testrun.ipynb) but not for **wflow**. We had to copy data from project directory to the user input directory, see `cell #6` in [02_wflow_testrun.ipynb](https://github.com/eWaterCycle/era5-comparison/blob/master/wflow/02_wflow_testrun.ipynb). For more information, see [wflow directory structure](https://wflow.readthedocs.io/en/latest/wflow_usage.html?highlight=directories#directory-structure-cases-and-runs).

**Comment by RolfHut** [comment]
I think this needs a general discussion on the nature and role of the input and output folder for all models in eWaterCycle. 

Shall we plan a short meeting / discussion on this? Or continue the discussion here?

**Comment by goord** [comment]
I can join if necessary

**Comment by SarahAlidoost** [comment]
A meeting discussion works better for me. @nielsdrost and @sverhoeven FYI.

**Comment by RolfHut** [comment]
agreed. IIRC, Niels wasn't working today, so shall we do this monday
towards the end of the day, say 16.00?

Rolf
-- 
PS my working hours may not be your working hours. Please only reply in
your working hours

*--*
*dr. ir. Rolf Hut*
MacGyver Assistant Professor in Water Resources at Delft University of
Technology
<http://www.tudelft.nl/>
Public speaker on Technology and Innovation for Everyone
Columnist at De Ingenieur <https://www.deingenieur.nl/artikel/quarantandem>
Writer of Rolfs Maakbare Wereld
<http://rolfhut.nl/rolfsmaakbarewereldbolcom> and Knotsgekke Surprises voor
Iedereen <http://rolfhut.nl/surpriseboek>.
Co-founder of Disdrometrics <http://disdro.com>.
www.rolfhut.nl/english
@RolfHut <http://twitter.com/@rolfhut>

*Recent papers*
Hut et al.: Easy to build low-power GPS drifters with local storage and a
cellular modem made from off-the-shelf components
<https://doi.org/10.5194/gi-9-435-2020>, Geoscientific Instrumentation,
Methods and Data Systems 2020
Hut et al.: Teaching a hands-on course during corona lockdown: from
problems to opportunities <https://doi.org/10.1088/1361-6552/abb06a>,
Physics Education 2020
Hut et al.: Taking a Breath of the Wild: are geoscientists more effective
than non-geoscientists in determining whether video game world landscapes
are realistic? <https://doi.org/10.5194/gc-2-117-2019>, Geoscience
Communication 2019

**Comment by SarahAlidoost** [comment]
I sent you a calendar invitation for the meeting title "Discuss grpc4bmi and in/output folders in eWaterCycle" on Monday.

**Comment by RolfHut** [comment]
this has become more urgent with more researchers working on the HPC machine with a shared input folder.

**Comment by sverhoeven** [comment]
Decided to 
* have multiple input directory mounted read-only (can be set writable as escape hatch)
* mount all dirs in same place as outside, see #93
* use work_dir for storing config file, see #93
* no more smart copying of config file

**Comment by sverhoeven** [comment]
0.2.12 has been released with this issues implemented. However 0.3.x / master branch has not been updated.
